### ISSUE TITLE
More user friendly UI

### ISSUE THREAD

**Issue by oscarrdg** [GEANT/CAT-Android#14]
The Android app UI is not very user friendly.

As there are actually some bad reviews on the play store complaining about it.
I.e:
https://play.google.com/store/apps/details?id=uk.ac.swansea.eduroamcat&reviewId=Z3A6QU9xcFRPRlY2X3ZZZnUyZy1tMVNBS3ppbngxeXFoOTVXb09oSkFfRENkX3JQNGJ5djRfQ2Y4S05MdnZ4TDJkV1cwbHpGR2hPZ1c0Ui0wajBIY1RseHc
https://play.google.com/store/apps/details?id=uk.ac.swansea.eduroamcat&reviewId=Z3A6QU9xcFRPRkc4Ymo4ZlR6RVg0UDc1RlZGLURzOW1Uc09LZEJZaDFmSHA1Wll2b2Z1QUhNMkpiSWM1X0tmMkhKay1CMU5maUJLT1lIWXBHVDdOMjA5VXc
https://play.google.com/store/apps/details?id=uk.ac.swansea.eduroamcat&reviewId=Z3A6QU9xcFRPRlE1TVBoV2Q4NTNYcUxmRldwc2xVMm1Nb2ZHTDZLdnE1NEVSUlgyMHZLdWhlRkQ1WXRWcnJ0c0JDT0tDNDItNXhFQnNmZGN0WUlBZHFRZ1E

I reckon most of the information shown on the UI right now are not necessary to a normal user. And it just make it look more complex than it really is.

EAP method, TTLS, PAP, MSCHAPv2, etc... all these words are confusing users as they do not understand what they mean.

A possibility to show "advanced information" would be nice to have, but it must not be the default.
The default user interaction with the app must be something as:

Open App --> Please select your Institution (showing closer institution available OR allowing the user to select from a list) --> Insert your personal credentials --> Installed!

Hiding by default every detail about the profile, the EAP method used, the certificate, ...

What do people think?
Maybe it is a known issue, but there is no resources to developed a streamlined ui?

Thank you a lot.

(Sorry if this is not the place to discuss this)

**Comment by GarethAyres** [comment]
Hello,

There is resource to change the UI, but there needs to be consensus on what is best. from Stefan and others in the development group.

I also feel removing some detail from the app would improve the usability for an average user, but advanced users may feel the information should always be visible for validation by users as a way to encourage best practise going forward. 

There is the argument that if the app doesnt demonstrate its programmatic validation of a profile for eduroam, then this is security concern and the app offers little over manual configuration. And if users dont validate a profile before installing it, this is another security concern. 

I certainly agree there is room to improve and streamline the UI, hopefully while keeping average and advanced users happy. It may be best to take this forward as a discussion in the eduroam development mailing list, Stefan?

Most the negative reviews in the play store are a result of Android caveats causing the app problems (cant programmatically remove existing profiles, must have a screen lock etc) which are out of our control sadly.

**Comment by oscarrdg** [comment]
@GarethAyres I agree with everything you said, but let's face that advanced users are just the 1% of the potential users of this app, so the default UI must be tailored to the 99% of the users, imo.

I also understand the security concern of installing something with validating it previously, but let's be honest: the "average user" will not be able to validate a profile, as they do not understand any word of it. The user will trust the app, and that's all.

**Comment by GarethAyres** [comment]
OK fair points. Lets see what @restena-sw thinks?

**Comment by restena-sw** [comment]
Well, yes, the app currently gives a "full disclosure" view with all technical details one could want. That's fine for a techie like me, but I see that this is not a view a majority appreciates.

I'm not a good UI designer myself though and don't think I can offer the best advice.

I'll ask around on the eduroam-devel mailing list if someone feels like doing UI work. If not, I can still make some suggestions of my own, and then maybe the OP could tell us how he likes those?

**Comment by dbiscalchin** [comment]
Hello,

I am a software developer at State University of Campinas, Brazil, and I'm also interested in improving the UX of this app. From my experience, I've noticed most users need help to set up eduroam here - they cannot do it by themselves.

I agree with @oscarrdg that the UI should be as simple as providing a username/password pair and clicking a button. In my opinion, more complex information should be hidden for the big majority of users. However, if there are people sticking with the idea that this information should be displayed all the time, I think it should be configurable through the app and hidden by default.

Also, the requirement for a screen lock is really bad. Thereby, if there isn't any better solution for that, I'd suggest making it easier, like opening the Android's screen lock settings automatically.

I'd like to help you with suggestions or even coding. So I'd be glad if you include me in the mailing list.

Thanks,

**Comment by restena-sw** [comment]
If you refer to the eduroam-devel mailing list? That's an open mailing list discussing all (technical) aspects of eduroam's future, with things like WPA3 or our upcoming EAP-TLS based eduroam Managed IdP product. If you only want to discuss the CAT app, that's only slightly matching the topic (but feel free to subscribe anyway: https://lists.eduroam.org/sympa/subscribe/development)

The better place to discuss would be cat-devel, which is again open and you can subscribe yourself: 
https://lists.geant.org/sympa/subscribe/cat-devel

Other than that, code contributions are of course welcome and should come in the form of Pull Requests.

**Comment by dbiscalchin** [comment]
I subscribed to both lists. Thank you, @restena-sw.
